src/mbgl/util/worker.cpp

```diff
@@ -0,0 +1,58 @@
+#include <mbgl/util/worker.hpp>
+
+#include <cassert>
+
+namespace mbgl {
+
+Worker::Worker(uv_loop_t* loop, std::size_t count, const char* name_)
+    : queue(new Queue(loop, [](std::function<void ()> after) { after(); }))
+    , name(name_)
+{
+    for (std::size_t i = 0; i < count; i++) {
+        threads.emplace_back(&Worker::thread, this);
+    }
+}
+
+Worker::~Worker() {
+    MBGL_VERIFY_THREAD(tid);
+
+    channel.send(Work());
+
+    for (auto& thread : threads) {
+        thread.join();
+    }
+
+    queue->stop();
+}
+
+void Worker::send(std::function<void ()> work,
+                  std::function<void ()> after) {
+    MBGL_VERIFY_THREAD(tid);
+    channel.send({work, after});
```

Comment:
There's a guard condition for `after` in the thread function. I'll add an assert for `work`.